test/e2e/autoscale_test.go

```diff
@@ -268,3 +268,44 @@ func TestAutoscaleUpDownUp(t *testing.T) {
 	assertScaleDown(ctx)
 	assertScaleUp(ctx)
 }
+
+func assertScaleUpCountPods(ctx *testContext) {
+	ctx.logger.Infof("The autoscaler spins up additional replicas when traffic increases.")
+	// note: without the warm-up / gradual increase of load the test is retrieving a 503 (overload) from the envoy
+	// options: https://istio.io/docs/concepts/traffic-management/#timeouts-and-retries
+	err := generateTraffic(ctx, 20, 30*time.Second)
+	if err != nil {
+		ctx.t.Fatalf("Error during initial scale up: %v", err)
+	}
+	AssertNumberOfPods(ctx, 2)
+	err = generateTraffic(ctx, 30, 30*time.Second)
+	if err != nil {
+		ctx.t.Fatalf("Error during initial scale up: %v", err)
+	}
+	AssertNumberOfPods(ctx, 3)
+	err = generateTraffic(ctx, 40, 30*time.Second)
+	if err != nil {
+		ctx.t.Fatalf("Error during initial scale up: %v", err)
+	}
+	AssertNumberOfPods(ctx, 4)
+}
+
+func AssertNumberOfPods(ctx *testContext, num_replicas int32) {
```

Comment:
I think you don't need to export this function (in go exporting means starting the name with an Uppercase letter). Lowercase should be fine just as well.